blobmodel/pulse_shape.py

```diff
@@ -0,0 +1,74 @@
+from abc import ABC, abstractmethod
+import numpy as np
+from nptyping import NDArray
+
+
+class AbstractBlobShape(ABC):
+    """Abstract class containing the blob pulse shapes. Two-dimensional blob
+    pulse shapes are written in the form.
+
+    phi(theta_x, theta_y) = phi_x(theta_x) * phi_y(theta_y).
+    """
+
+    @abstractmethod
+    def get_pulse_shape_prop(self, theta_prop: NDArray, kwargs):
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_pulse_shape_perp(self, theta_perp: NDArray, kwargs):
+        raise NotImplementedError
+
+
+class BlobShapeImpl(AbstractBlobShape):
+    """Implementation of the AbstractPulseShape class."""
+
+    __SHAPE_NAMES__ = {"exp", "gauss", "2-exp", "lorentz", "secant"}
+
+    def __init__(self, pulse_shape_prop="gauss", pulse_shape_perp="gauss"):
+        self.__GENERATORS__ = {
```

Comment:
I haven't seen this syntax before, is it basically a match - case block that instantiates the matching object?

Reply:
It is a dictionary